### PR TITLE
[Docs] Add detail on using temporary tables in Cloud

### PR DESCRIPTION
**Pull request by justindeguzman**
### Changelog category (leave one):
- Documentation (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add detail on using temporary tables in Cloud

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 5de010ddda8311ff178519c0161b6d8314e2641e with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59612/5de010ddda8311ff178519c0161b6d8314e2641e/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

**Review by melvynator — COMMENTED** [review]
It's also important to highlight that temporary tables are dropped at the end of the session.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by justindeguzman** [comment]
> It's also important to highlight that temporary tables are dropped at the end of the session.

We actually already say this just right below the note. "Temporary tables disappear when the session ends, including if the connection is lost."

Does it make sense to rewrite the note to include the above for emphasis?

File: docs/en/sql-reference/statements/create/table.md [hunk 1]
```diff
@@ -514,6 +514,10 @@ ENGINE = MergeTree ORDER BY x;
 
 ## Temporary Tables
 
+:::note
+Please note that on ClickHouse Cloud, temporary tables are not replicated. As a result, there is no guarantee that data inserted into a temporary table will be available in other replicas. The primary use case where temporary tables can be useful in ClickHouse Cloud is for querying or joining small external datasets during a single session.
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/statements/create/table.md:518]:
```suggestion
Please note that temporary tables are not replicated. As a result, there is no guarantee that data inserted into a temporary table will be available in other replicas. The primary use case where temporary tables can be useful is for querying or joining small external datasets during a single session.
```

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**